sunbeam-python/sunbeam/plugins/interface/v1/openstack.py

```diff
@@ -0,0 +1,290 @@
+# Copyright (c) 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import inspect
+import logging
+import shutil
+from abc import abstractmethod
+from enum import Enum
+from pathlib import Path
+from typing import Optional
+
+import click
+from packaging.version import Version
+from rich.console import Console
+from rich.status import Status
+from snaphelpers import Snap
+
+from sunbeam.clusterd.client import Client
+from sunbeam.clusterd.service import ConfigItemNotFoundException
+from sunbeam.commands.juju import JujuStepHelper
+from sunbeam.commands.openstack import (
+    OPENSTACK_MODEL,
+    determine_target_topology_at_bootstrap,
+)
+from sunbeam.commands.terraform import (
+    TerraformException,
+    TerraformHelper,
+    TerraformInitStep,
+)
+from sunbeam.jobs.checks import VerifyBootstrappedCheck
+from sunbeam.jobs.common import (
+    BaseStep,
+    Result,
+    ResultType,
+    read_config,
+    run_plan,
+    run_preflight_checks,
+    update_config,
+)
+from sunbeam.jobs.juju import JujuHelper, JujuWaitException, TimeoutException, run_sync
+from sunbeam.plugins.interface.v1.base import EnableDisablePlugin
+
+LOG = logging.getLogger(__name__)
+console = Console()
+
+APPLICATION_DEPLOY_TIMEOUT = 1200  # 15 minutes
+
+
+class TERRAFORM_PLAN_LOCATION(Enum):
```

Comment:
```suggestion
class TerraformPlanLocation(Enum):
```

Pep8 states enums should be named like classes